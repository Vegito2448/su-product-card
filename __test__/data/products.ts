const product1 = {
  id: '1',
  title: 'Coffee Mug',
  description: 'A mug for your coffee',
  img: '/coffee-mug.png',
  price: 10,
};
const product2 = {
  id: '1',
  title: 'Coffee Mug without Image',
  description: 'A mug for your coffee',
  img: '/no-image.jpg',
  price: 10,
};
export { product1, product2 };
