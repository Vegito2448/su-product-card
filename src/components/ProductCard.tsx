import React, { CSSProperties, createContext } from "react";
import useProduct from "../hooks/useProduct";
import { Product, ProductCardHandlers, ProductContextProps, initialValues, onChangeProductArgs } from '../interfaces/types';
import styles from '../styles/styles.module.css';
export interface ProductCardProps {
  product: Product;
  // children?: ReactElement | ReactElement[];
  children: (args: ProductCardHandlers) => JSX.Element;
  className?: string;
  style?: CSSProperties;
  onChange?: (args: onChangeProductArgs) => void;
  value?: number;
  initialValues?: initialValues;
}

export const ProductContext = createContext({} as ProductContextProps);

const { Provider: ProductProvider } = ProductContext;

const ProductCard = ({ product, children, className, style, onChange, value, initialValues }: ProductCardProps) => {

  const { counter,
    increaseBy,
    maxCount,
    reset,
    isMaxReached
  } = useProduct({ onChange, product, value, initialValues });

  return (
    <ProductProvider
      value={{ counter, increaseBy, product, maxCount }}
    >
      <div className={`${styles.productCard} ${className}`}
        style={style}
      >
        {children({
          quantity: counter,
          isMaxReached,
          maxCount,
          product,
          increaseBy,
          reset
        })}
        {/* <ProductImage img={product.img} /> */}

        {/* <ProductTitle title={product.title} /> */}

        {/* <ProductButtons counter={counter} increaseBy={increaseBy} /> */}
      </div>
    </ProductProvider>
  );
};

export default ProductCard;