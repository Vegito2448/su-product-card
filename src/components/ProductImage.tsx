import React, { CSSProperties, useContext } from "react";
import noImage from '../assets/no-image.jpg';
import styles from '../styles/styles.module.css';
import { ProductContext } from "./ProductCard";
export interface ProductImageProps {
  className?: string;
  img?: string;
  style?: CSSProperties;
}


export const ProductImage = ({ img, className, style }: ProductImageProps) => {
  const { product: { img: imgProvider } } = useContext(ProductContext);

  img = img || imgProvider;

  return <img
    src={img || noImage} alt="Product"
    className={`${styles.productImg} ${className}`}
    style={style}
  />;
};