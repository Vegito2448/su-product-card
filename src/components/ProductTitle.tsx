

import React, { CSSProperties, useContext } from "react";
import styles from '../styles/styles.module.css';
import { ProductContext } from "./ProductCard";
export interface ProductTitleProps {
  className?: string;
  title?: string;
  style?: CSSProperties;
}

export const ProductImage = ({ title, className, style }: ProductTitleProps) => {

  const { product: { title: titleProvider = '' } } = useContext(ProductContext);

  title = title || titleProvider || '';

  return <span
    className={`${styles.productDescription} ${className || ''}`}
    style={style}
  >{title}</span>;
};