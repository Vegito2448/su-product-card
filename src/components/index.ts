import { ProductCardHocProps } from "../interfaces/types";
import { ProductButtons } from "./ProductButtons";
import ProductCardHoc from "./ProductCard";
import { ProductImage } from "./ProductImage";

const ProductCard: ProductCardHocProps = Object.assign(ProductCardHoc, {
  Title: ProductImage,
  Image: ProductImage,
  Buttons: ProductButtons,
});


export {
  ProductButtons,
  ProductCard,
  ProductImage,
  ProductImage as ProductTitle
};

