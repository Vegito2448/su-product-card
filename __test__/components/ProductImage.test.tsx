import React from "react";
import render from "react-test-renderer";
import { ProductCard, ProductImage } from '../../src/components';
import { product1 } from "../data";

describe('ProductImage', () => {

  test('should render the product image with the default data', () => {
    const wrapper = render.create(
      <ProductCard
        product={product1}
      >
        {() =>
          <ProductImage

          />}
      </ProductCard>);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});