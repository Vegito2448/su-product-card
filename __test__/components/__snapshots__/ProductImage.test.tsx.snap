// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should render the product image with the default data 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="/coffee-mug.png"
  />
</div>
`;
